Source/Wikitivity.Agent/WikitivityAgentHelper.cs

```diff
@@ -196,55 +322,42 @@ public static void ImportJobOnComplete(JobReport jobReport)
 			IAPIMessages += jobReport.ToString();
 		}
 
-		public List<int> GetCasesWithWikitivity(IRSAPIClient proxy)
+		
+
+		public async Task<List<int>> GetCasesWithWikitivityOM(IObjectManager proxy)
 		{
+			//TODO: Refactor to OM
 			List<int> CasesWithWikitivity = new List<int>();
 
-			Query<Workspace> getCaseIDs = new Query<Workspace>();
-			getCaseIDs.Condition = new WholeNumberCondition("Artifact ID", NumericConditionEnum.IsSet);
-			getCaseIDs.Fields = FieldValue.AllFields;
-			QueryResultSet<Workspace> CaseQueryResults = new QueryResultSet<Workspace>();
-			proxy.APIOptions.WorkspaceID = -1;
+			QueryRequest getWorkspaceList = new QueryRequest()
+			{
+				ObjectType = new ObjectTypeRef() { Name = "Workspace" },
+				Condition = "('ArtifactID' ISSET)",
+				IncludeIDWindow = false,
+				RelationalField = null, 
+				SampleParameters = null,
+				SearchProviderCondition = null,
+				Sorts = null,
+				Fields = new List<FieldRef>()
+				{
+					new FieldRef() {Name = "ArtifactID"},
+				}
+			};
+
 			try
 			{
-				CaseQueryResults = proxy.Repositories.Workspace.Query(getCaseIDs, 0);
+				Relativity.Services.Objects.DataContracts.QueryResult queryResults = await proxy.QueryAsync(-1, getWorkspaceList, 0, 10000);

```

Comment:
I have see that 10,000 number at least twice so far in the review. I would move it to a Constants class and then re-use it everywhere.